Public/Src/Cache/ContentStore/Distributed/NuCache/LocalLocationStore.cs

```diff
@@ -600,17 +600,20 @@ private async Task<BoolResult> RestoreCheckpointStateAsync(OperationContext cont
             {
                 var latestCheckpoint = _checkpointManager.GetLatestCheckpointInfo(context);
                 var latestCheckpointAge = _clock.UtcNow - latestCheckpoint?.checkpointTime;
-                var shouldRestoreInBackground = latestCheckpointAge < _configuration.Checkpoint.RestoreCheckpointAgeThreshold;
+
+                // Only skip if this is the first restore and it is sufficiently recent
+                var shouldSkipRestore = _lastRestoreTime == default 
+                    && latestCheckpoint != null 
+                    && latestCheckpoint?.checkpointTime.IsRecent(_clock.UtcNow, _configuration.Checkpoint.RestoreCheckpointAgeThreshold) == true;

```

Comment:
very nit: either 'lastCheckpont != null' or the question mark in 'lastCheckpoint?.checkpointTime' are redundant.